### ISSUE TITLE
IndexOutOfBoundsException

### ISSUE THREAD

**Issue by MaAnQing** [PhilJay/MPAndroidChart#2952]
``` java
java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at java.util.ArrayList.get(ArrayList.java:308)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at com.github.mikephil.charting.data.DataSet.getEntryForIndex(DataSet.java:286)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at com.github.mikephil.charting.utils.Transformer.generateTransformedValuesLine(Transformer.java:184)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at com.github.mikephil.charting.renderer.LineChartRenderer.drawValues(LineChartRenderer.java:545)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at com.github.mikephil.charting.charts.BarLineChartBase.onDraw(BarLineChartBase.java:251)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:16224)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.updateDisplayListIfDirty(View.java:15221)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:15994)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.drawChild(ViewGroup.java:3611)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3401)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:16227)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.updateDisplayListIfDirty(View.java:15221)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:15994)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.drawChild(ViewGroup.java:3611)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3401)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.updateDisplayListIfDirty(View.java:15216)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:15994)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.drawChild(ViewGroup.java:3611)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3401)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.updateDisplayListIfDirty(View.java:15216)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:15994)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.drawChild(ViewGroup.java:3611)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3401)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.updateDisplayListIfDirty(View.java:15216)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:15994)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.drawChild(ViewGroup.java:3611)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3401)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.updateDisplayListIfDirty(View.java:15216)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:15994)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.drawChild(ViewGroup.java:3611)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3401)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.updateDisplayListIfDirty(View.java:15216)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:15994)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.drawChild(ViewGroup.java:3611)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3401)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.updateDisplayListIfDirty(View.java:15216)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:15994)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.drawChild(ViewGroup.java:3611)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3401)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.updateDisplayListIfDirty(View.java:15216)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:15994)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.drawChild(ViewGroup.java:3611)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3401)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.updateDisplayListIfDirty(View.java:15216)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:15994)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.drawChild(ViewGroup.java:3611)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewGroup.dispatchDraw(ViewGroup.java:3401)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.draw(View.java:16227)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at com.android.internal.policy.PhoneWindow$DecorView.draw(PhoneWindow.java:3056)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.View.updateDisplayListIfDirty(View.java:15221)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ThreadedRenderer.updateViewTreeDisplayList(ThreadedRenderer.java:296)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ThreadedRenderer.updateRootDisplayList(ThreadedRenderer.java:302)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ThreadedRenderer.draw(ThreadedRenderer.java:337)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewRootImpl.draw(ViewRootImpl.java:2853)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewRootImpl.performDraw(ViewRootImpl.java:2670)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2300)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1323)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6339)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.Choreographer.doCallbacks(Choreographer.java:670)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.Choreographer.doFrame(Choreographer.java:606)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
03-22 21:34:37.714 E/AndroidRuntime( 6445): 	at android.os.Handler
```

**Comment by MGaetan89** [comment]
This is a duplicate of #2450

**Comment by almic** [comment]
fixed now in d67ea481af1e8528c617dfd404d0c7827b0134b2
